extensions/multi/src/org/sbml/jsbml/validator/offline/constraints/MultiModelPluginConstraints.java

```diff
@@ -90,10 +93,11 @@ public void addErrorCodesForAttribute(Set<Integer> set, int level,
       func = new UnknownAttributeValidationFunction<MultiModelPlugin>() {
         @Override
         public boolean check(ValidationContext ctx, MultiModelPlugin multiM) {
-
-          // TODO
+        	
           // There may be at most one ListOfSpeciesTypes container object within a Model object
-          return true;
+          ValidationFunction<SBMLDocument> f = new DuplicatedElementValidationFunction<SBMLDocument>(MultiConstants.listOfSpeciesTypes);
+             
+          return f.check(ctx, multiM.getParent());
```

Comment:
Here for the check, you are using multiM.getParent() which is not correct. It could work like that but we want to make it work on multiM directly. Because the parent will be the Model and we want to store the data on the MultiModelPlugin instance.
To have the DuplicatedElementValidationFunction check to work (check the source code of this class), you need to modify the MultiParser class to add the XML elements as they are read into a list stored in the user objects that each SBase can have.

On https://github.com/sbmlteam/jsbml/blob/361d88121d8781709abd40b04f31006b0f77b336/extensions/multi/src/org/sbml/jsbml/xml/parsers/MultiParser.java#L180, you need to add

// keep order of elements for later validation
AbstractReaderWriter.storeElementsOrder(elementName, multiModel);

After that, the check should be nicely implemented and working fine.

Reply:
I'll try to understand this. I'm having a bit of trouble getting this right now.

Reply:
You should check the implementation of AbstractReaderWriter.storeElementsOrder and DuplicatedElementValidationFunction.

Reply:
I went through the codes of both ```AbstractReaderWriter.storeElementsOrder``` and ```DuplicatedElementValidationFunction```, and as of what I understand, they are broadly serving the same purpose, of adding the elements to a list. 
The reason for adding elements to a list in ```MultiParser``` , according to me is that the parser fills the list with the elements relating to the specific extension and when ```DuplicatedElementValidationFunction``` is used, it gets the elements stored in the specific list of the model plugin of that extension.
Is this correct?

Reply:
Exactly. While we read the XML file, the elements are sent to the parser that will create the object instances and fill in the attributes and children. While the parser is doing that, we can store the elements in the order they are encountered which allow us to to the validation later on.